vision/detect.js

```diff
@@ -0,0 +1,457 @@
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const Vision = require('@google-cloud/vision');
+const Storage = require('@google-cloud/storage');
+
+// [START vision_face_detection]
+function detectFaces (fileName) {
+  // Instantiates a client
+  const vision = Vision();
+
+  // The local image file on which to perform face detection
+  // fileName = '/path/to/image.jpg';
+
+  // Performs face detection on the local file
+  return vision.detectFaces(fileName)
+    .then((results) => {
+      const faces = results[0];
```

Comment:
Nit. Prefer [four spaces on breaks when wrapping preceding line](https://google.github.io/styleguide/javascriptguide.xml?showone=Code_formatting#Code_formatting):

```
  return vision.detectFaces(fileName)
      .then((results) => { 
        const faces = results[0];
```

That said, your aforementioned two-space style persists throughout and is consistent with anonymous function spacing so you're free to ignore as this style is consistent elsewhere and would require more changes to remain so, which is more important.